addons/mrp_repair/models/mrp_repair.py

```diff
@@ -0,0 +1,539 @@
+    invoice_id = fields.Many2one('account.invoice', string='Invoice', readonly=True, track_visibility="onchange", copy=False)
+    move_id = fields.Many2one('stock.move', string='Move', readonly=True, help="Move created by the repair order", track_visibility="onchange", copy=False)
+    fees_lines_ids = fields.One2many('mrp.repair.fee', 'repair_id', string='Fees', readonly=True, states={'draft': [('readonly', False)]}, copy=True, oldname="fees_lines")
+    internal_notes = fields.Text(string='Internal Notes')
+    quotation_notes = fields.Text(string='Quotation Notes')
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env['res.company']._company_default_get('mrp.repair'))
+    invoiced = fields.Boolean(string='Invoiced', readonly=True, copy=False)
+    repaired = fields.Boolean(string='Repaired', readonly=True, copy=False)
+    amount_untaxed = fields.Float(string='Untaxed Amount', compute='_amount_untaxed', store=True)
+    amount_tax = fields.Float(string='Taxes', compute='_amount_tax', store=True)
+    amount_total = fields.Float(string='Total', compute='_amount_total', store=True)
+
+    _sql_constraints = [('name', 'unique (name)', 'The name of the Repair Order must be unique!'), ]
+
+    @api.onchange('product_id')
+    def onchange_product_id(self):
+        """ On change of product sets some values.
+        @param product_id: Changed product
+        @return: Dictionary of values.
+        """
+        if self.product_id:
+            self.guarantee_limit = False
+            self.lot_id = False
+            self.product_uom_id = self.product_id.uom_id.id
+
+    @api.onchange('product_uom_id')
+    def onchange_product_uom_id(self):
+        res = []
+        if not self.product_uom_id or not self.product_id:
+            return res
+        if self.product_uom_id.category_id.id != self.product_id.uom_id.category_id.id:
+            raise UserError('The Product Unit of Measure you chose has a different category than in the product form.')
+            self.product_uom_id = self.product_id.uom_id.category_id.id
+
+    @api.onchange('location_id')
+    def onchange_location_id(self):
+        """ On change of location
+        """
+        self.location_dest_id = self.location_id.id
+
+    @api.multi
+    def button_dummy(self):
+        return True
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        """ On change of partner sets the values of partner address,
+        partner invoice address and pricelist.
+        @param part: Changed id of partner.
+        @param address_id: Address id from current record.
+        """
+        if not self.partner_id:
+            self.address_id = False
+            self.partner_invoice_id = False
+            self.pricelist_id = self.pricelist_id.search([('type', '=', 'sale')], limit=1).id
```

Comment:
Actually, filtered will not work here, I used self.pricelist_id instead of fetching product.pricelist from pool:
```self.env['product.pricelist'].search([('type', '=', 'sale')], limit=1).id```